service.go

```diff
@@ -240,7 +240,7 @@ func (s *Service) Stop(force bool) error {
 // version number, config profile, config directory, whether to use registry, and Driver, which cannot be nil.
 // Note - this function is a singleton, if called more than once,
 // it will always return an error.
-func NewService(serviceName string, serviceVersion string, confProfile string, confDir string, useRegistry string, proto dsModels.ProtocolDriver) (*Service, error) {
+func NewService(serviceName string, serviceVersion string, confProfile string, confDir string, useRegistry string, proto dsModels.ProtocolDriver, overwriteConfig bool) (*Service, error) {
```

Comment:
isn't this a breaking change since we expose this API?

Reply:
thanks for noticing that.  I made a workaround now although it's not perfect.

Reply:
Requiring a device service author to include an internal global variable package is not acceptable. Frankly I'm not sure why we're using global variables in the sdk in the first place. The original implementation had many of these vars as part of the service struct. What happens if someone calls NewService twice? Is this supported?

Anyways, back to this PR itself. A much better approach would be to add a new Service method called SetConfigOverwrite (or something like that) such that you'd create the new service, then call this method to configure the service to overwrite the consul config, then you'd call Start.

We should probably also file an issue for Geneva about re-factoring the API based on feedback from device service developers that have used the 1.x API.

Reply:
No, it doesn't support calling NewService twice, and it would log warning and return an error.
The reason I adopted global variable because I use packages to separate the logic instead of the flat package.  If we keep them in the Service struct, it would cause the circulated dependency compile error.  For example, Service uses config package to load configuration and config package needs to access the variable of Service struct, and the compile error occurs. 

We could keep opening a series of issues about re-factoring the API based on feedback.

Reply:
LoadConfig behavior is in NewService, so we can't call Service.SetOverwriteConfig(bool) after that.